### PR TITLE
greying out disabled editText

### LINKED ISSUE #8247: [Bug] Changing note type in note editor make fields non clickable
###### Reproduction Steps

1. Open deck
2. Edit note
3. Change note type
4. Click on fields, it is not clickable
5. Change type to original note type now fields are clickable.

###### Expected Result
It should make other fields clickable in note editor when note type change.

###### Actual Result
It is not making fields clickable.
</img>

###### Debug info
Refer to the [support page](https://ankidroid.org/docs/help.html) if you are unsure where to get the "debug info".
AnkiDroid Version = 2.15alpha30

Android Version = 11

ACRA UUID = c36be804-6080-4d1b-a9e7-5633cd4703fe

Scheduler = std

Crash Reports Enabled = true

DatabaseV2 Enabled = true

Model = Samsung M30s

###### Research
*Enter an [x] character to confirm the points below:*

- [ x ] I have read the [support page](https://ankidroid.org/docs/help.html) and am reporting a bug or enhancement request specific to AnkiDroid
- [ x ] I have checked the [manual](https://ankidroid.org/docs/manual.html) and the [FAQ](https://github.com/ankidroid/Anki-Android/wiki/FAQ) and could not find a solution to my issue
- [ x ] I have searched for similar existing issues here and on the user forum
- [ x ] (Optional) I have confirmed the issue is not resolved in the latest alpha release ([instructions](https://docs.ankidroid.org/manual.html#betaTesting))

### PR DESCRIPTION
**Pull request by prateek-singh-3212**
## Purpose / Description
Changing note type in note editor make fields non-clickable

## Fixes
Fixes #8247 

## Approach
Greyed out the edit text fields when edit text fields are disabled

## How Has This Been Tested?

Tested on Samsung Galaxy M51, android 11 (Samsung One UI 3.1)

## Checklist
_Please, go through these checks before submitting the PR._

- [x] You have not changed whitespace unnecessarily (it makes diffs hard to read)
- [x] You have a descriptive commit message with a short title (first line, max 50 chars).
- [x] Your code follows the style of the project (e.g. never omit braces in `if` statements) 
- [x] You have commented your code, particularly in hard-to-understand areas
- [x] You have performed a self-review of your own code
- [x] UI changes: include screenshots of all affected screens (in particular showing any new or changed strings)
- [x] UI Changes: You have tested your change using the [Google Accessibility Scanner](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor)

### CODE REVIEW

File: AnkiDroid/src/main/java/com/ichi2/anki/NoteEditor.java [hunk 1]
```diff
@@ -1757,6 +1758,10 @@ private void initFieldEditText(FieldEditText editText, final int index, boolean
                 }
             });
         }
+
+        if(!enabled){
```
Comment (david-allison-1) [inline AnkiDroid/src/main/java/com/ichi2/anki/NoteEditor.java:1762]:
```suggestion
        if (!enabled){
```

File: AnkiDroid/src/main/java/com/ichi2/anki/NoteEditor.java [hunk 2]
```diff
@@ -1757,6 +1758,10 @@ private void initFieldEditText(FieldEditText editText, final int index, boolean
...
+            editText.setBackgroundColor(Color.parseColor("#CFCFCF"));
```
Comment (david-allison-1) [inline AnkiDroid/src/main/java/com/ichi2/anki/NoteEditor.java:1763]: Could you extract this to a resource (for example: https://github.com/ankidroid/Anki-Android/blob/8fa3e45e2d3cf4d9898caf990825429f472faad9/AnkiDroid/src/main/res/values/theme_light.xml#L14)

**Review by david-allison-1 — CHANGES_REQUESTED** [review]
Looks good! A few tiny changes, then good to go, thanks!

PS: Could you also test this in night mode and make sure it looks reasonable

File: AnkiDroid/src/main/java/com/ichi2/anki/NoteEditor.java (hunk 2 above)
Comment (prateek-singh-3212) [reply AnkiDroid/src/main/java/com/ichi2/anki/NoteEditor.java:1763]: Yes,
here is the screenshot of both light and dark mode and I have linked this with color.xml. Should I use different colors for both light and dark modes?
WhatsApp Image 2021-03-25 at 19 47 43(1)
WhatsApp Image 2021-03-25 at 19 47 43

Comment (david-allison-1) [reply AnkiDroid/src/main/java/com/ichi2/anki/NoteEditor.java:1763]: I trust your judgment - it should be easy to do if extracted to `resources`, see the `theme-` xml files.

File: AnkiDroid/src/main/java/com/ichi2/anki/NoteEditor.java [hunk 3]
```diff
@@ -1757,6 +1758,10 @@ private void initFieldEditText(FieldEditText editText, final int index, boolean
                 }
             });
         }
+
+        if (!enabled){
```
Comment (david-allison-1) [inline AnkiDroid/src/main/java/com/ichi2/anki/NoteEditor.java:1762]: Really sorry 😅

```suggestion
        if (!enabled) {
```

File: AnkiDroid/src/main/java/com/ichi2/anki/NoteEditor.java (hunk 2 above)
Comment (prateek-singh-3212) [reply AnkiDroid/src/main/java/com/ichi2/anki/NoteEditor.java]: Apology for the late work. I have added different colors for different themes and liked their code with theme files. 
Here is the screenshot of both light and dark theme
WhatsApp Image 2021-03-26 at 09 58 28(1)
WhatsApp Image 2021-03-26 at 09 58 28

**Review by david-allison-1 — APPROVED** [review]
Looks good to me, thanks!

**Comment by david-allison-1** [comment]
In future, please don't perform a `merge` commit into your branch, instead, rebase it onto `master`

**Comment by prateek-singh-3212** [comment]
> In future, please don't perform a `merge` commit into your branch, instead, `rebase` it onto `master`

okay, I will take care of this thing further.

**Review by mikehardy — APPROVED** [review]
LGTM thanks!

**Pull request merged**